// echo.proto
syntax = "proto3";
package display;

// The request message containing parameters.
message LaunchRequest {
  string ControlSignalEndpoint_STAGE = 1;
  string CodeMappingEndpoint_STAGE = 2;
  string SignalingServer = 3;
  string processName = 4;
}

message TerminateRequest {
  string processName = 1;
  string subProcessName = 2;
}

// The response message containing success/failure message.
message Reply {
  string result = 1;
  string message = 2;
}

// The echo service definition.
service Launch {
  // Launch reply.
  rpc Start (LaunchRequest) returns (Reply) {}
  rpc Terminate (TerminateRequest) returns (Reply) {}
}